src/main/java/org/ballerinax/datamapper/diagnostic/DataMapperDiagnosticLog.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinax.datamapper.diagnostic;
+
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticCode;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+import io.ballerina.tools.diagnostics.Location;
+import org.ballerinalang.model.elements.PackageID;
+import org.ballerinalang.util.diagnostic.DiagnosticLog;
+import org.wso2.ballerinalang.compiler.PackageCache;
+import org.wso2.ballerinalang.compiler.diagnostic.BLangDiagnostic;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+
+import java.text.MessageFormat;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+/**
+ * Diagnostic log for Data Mapper Compiler Extension.
+ */
+public class DataMapperDiagnosticLog implements DiagnosticLog {
+    private static final CompilerContext.Key<DataMapperDiagnosticLog> DIAGNOSTIC_LOG_KEY = new CompilerContext.Key();
+    private static final String ERROR_PREFIX = "error";
+    private static final ResourceBundle MESSAGES = ResourceBundle.getBundle("datamapper", Locale.getDefault());
+
+    private int errorCount = 0;
+    private PackageCache packageCache;
+    private PackageID currentPackageId;
+    private boolean isMute = false;
+
+    public DataMapperDiagnosticLog(CompilerContext context) {
```

Comment:
If the idea is to make this logger a per-context logger, the pattern we use is, making the constructor public and use the static `getInstance()` method to retrieve a logger instance associated with a particular context. See https://github.com/ballerina-platform/ballerina-lang/blob/70e126663d6969b2c63748684f5b899c24424e54/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLog.java#L53 for example

Reply:
Ok, Fixed in [3749d63](https://github.com/ballerina-platform/module-ballerinax-datamapper/pull/36/commits/3749d6315a93bed2c289bd683bbf40f8d541089a)